packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,116 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+
+export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** If the link will redirect users to a page in or out Grafana */
+  externalLink?: boolean;
```

Comment:
Could this be just `external`?

Reply:
We should also mention that if this is true a little icon will appear next to it unless the `icon` prop has a different icon

Reply:
At first, using `external` showed it as 'deprecated' as it linked it to another context, but now it looks ok so I change it.
Should we show an icon just with `external` set to `true` or would we need `inline` to be `false` too? If we don't clarify it we could set showing the icon as default when the link is surrounded by more text, and this may be not wanted. Wdyt?

Reply:
This is sort of related to this other comment I made https://github.com/grafana/grafana/pull/69330#discussion_r1214238835 I agree that if the functionality is only to show when inline is false and external is true we should mention that as well 👍